### PR TITLE
Update to odo 2.0.3

### LINKED ISSUE #227: Switch to odo 2.0
[kind/Feature]

## Which functionality do you think we should add?

Switch to odo 2.0

## Why is this needed?

This is a major milestone and bring new features (devfile,....)

[ ] #228 
[ ] #229

### PR DESCRIPTION
**Pull request by sbouchet**
## What is the purpose of this change? What does it change?
update odo to latest 2.0.3

## Was the change discussed in an issue?
fixes #227

### CODE REVIEW

**Comment by sbouchet** [comment]
@jeffmaury actions are messy but tests results are expected ( 1 falling due to link service with devfile )

File: src/main/java/org/jboss/tools/intellij/openshift/actions/component/LinkServiceAction.java [hunk 1]
```diff
@@ -76,7 +79,7 @@ public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object s
        } else {
           UIHelper.executeInUI(() -> Messages.showWarningDialog("No services to link to", "Link service"));
         }
-      } catch (IOException e) {
+      } catch (IOException | KubernetesClientException e) {
```
Comment (jeffmaury) [inline src/main/java/org/jboss/tools/intellij/openshift/actions/component/LinkServiceAction.java:82]: Not sure why we need this exception but the patter is that the Odo interface throws IOExceptions and implementations should translate KubernetesClientException  into IOException

**Review by jeffmaury — CHANGES_REQUESTED** [review]

File: src/main/java/org/jboss/tools/intellij/openshift/actions/component/LinkServiceAction.java (hunk 1 above)
Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/actions/component/LinkServiceAction.java:82]: I was getting this exception when trying to link services to components and it wasn't catch. so user was facing a no op whane clicking on the menu. I'll improve to catch only IOE.

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java [hunk 2]
```diff
@@ -478,9 +478,13 @@ public void logout() throws IOException {
...
-    public List<ServiceInstance> getServices(String project, String application) {
-        ServiceCatalogClient sc = client.adapt(ServiceCatalogClient.class);
-        return sc.serviceInstances().inNamespace(project).withLabelSelector(new LabelSelectorBuilder().addToMatchLabels(KubernetesLabels.APP_LABEL, application).build()).list().getItems();
+    public List<ServiceInstance> getServices(String project, String application) throws IOException {
+        try {
+            ServiceCatalogClient sc = client.adapt(ServiceCatalogClient.class);
+            return sc.serviceInstances().inNamespace(project).withLabelSelector(new LabelSelectorBuilder().addToMatchLabels(KubernetesLabels.APP_LABEL, application).build()).list().getItems();
+        } catch (KubernetesClientException e) {
+            throw new IOException("Error when getting services list: "+e.getLocalizedMessage(), e);
+        }
```
Comment (sbouchet) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:487]: @jeffmaury this new implementation breaks tree node UI in https://github.com/redhat-developer/intellij-openshift-connector/blob/00a7b83a7eb03993b5d233313bf91cf071a811b0/src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationNode.java#L40

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java:487]: reverted and return empty list. in latest commit.

**Review by jeffmaury — APPROVED** [review]

**Pull request merged**